### PR TITLE
Nginx: new packages, master process reload and various fixes

### PR DESCRIPTION
**Pull request by dpausp**
* update default nginx to 1.18, add 1.19
* reload master process on package changes instead of restarting
* make worker age sensu check less noisy
* add missing modsecurity module
* do not listen on 0.0.0.0 by default
* fix broken NixOS option link

@flyingcircusio/release-managers

## Release process

Impact:

* [NixOS 19.03] Nginx will be restarted.

Changelog:

* Nginx (#127883)

  * Update default package to latest stable version 1.18.0, add mainline version 1.19.0
  * Structured JSON config: use frontend IPs instead of all interfaces when listenAddress is not given.
  * Reload master process when Nginx package changes. This avoids restarts for updates in the future.
  * Add modsecurity module.
  * Don't listen on 0.0.0.0 port 80 by default.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - use recent versions
  - secure defaults: don't listen on all interfaces by default
    - only localhost without config
    - structured config uses frontend IPs by default
  - version updates should not introduce changes that compomise security
- [x] Security requirements tested? (EVIDENCE)
  - 1.18 is the newest stable version, 1.19.0 is not the current version but quite recent
  - checked changelog for security-related changes: http://nginx.org/en/CHANGES-1.18 http://nginx.org/en/CHANGES
  - manual checks on a test VM that Nginx listens only on the expected addresses/interfaces
  - automatic test checks basic functionality and proper reloading

### CODE REVIEW

File: nixos/services/nginx/base-module.nix [hunk 1]
```diff
@@ -304,6 +304,29 @@ let
...
+            echo "Stopping old nginx master process..."
+            ${pkgs.procps}/bin/pkill -QUIT -F /run/nginx/nginx.pid.oldbin
+            echo "Nginx master process replacement complete"
+            exit 0
+        fi
+    done
+
+    echo Warning: new master process did not start.
+    echo Check journalctl -eu nginx and run systemctl restart nginx to activate changes.
+  '';
```
Comment (dpausp) [inline nixos/services/nginx/base-module.nix]: exit 1 on failure

File: nixos/services/nginx/base-module.nix [hunk 2]
```diff
@@ -304,6 +304,29 @@ let
   );
 
   checkConfigCmd = "${cfg.package}/bin/nginx -t -c ${configFile} -p ${cfg.stateDir}";
+
+  nginxReloadMaster = pkgs.writeScriptBin "nginx-reload-master" ''
+    set -e
+    echo "Starting new nginx master process..."
+    ${pkgs.procps}/bin/pkill -USR2 -F /run/nginx/nginx.pid
```
Comment (dpausp) [inline nixos/services/nginx/base-module.nix]: let

File: nixos/services/nginx/base-module.nix [hunk 3]
```diff
@@ -304,6 +304,29 @@ let
...
+            echo "Stopping old nginx workers..."
+            ${pkgs.procps}/bin/pkill -WINCH -F /run/nginx/nginx.pid.oldbin
+            echo "Stopping old nginx master process..."
+            ${pkgs.procps}/bin/pkill -QUIT -F /run/nginx/nginx.pid.oldbin
+            echo "Nginx master process replacement complete"
+            exit 0
+        fi
+    done
+
+    echo Warning: new master process did not start.
```
Comment (dpausp) [inline nixos/services/nginx/base-module.nix]: quote

**Review by ckauhaus — APPROVED** [review]
reviewed code on jitsi

**Pull request merged**